docs/plugins/pluggable_authentication.md

```diff
@@ -1,9 +1,201 @@
 #
 
+## Legacy Native Authentication
+
 By default, iRODS uses a secure password system for user authentication.  The user passwords are scrambled and stored in the iCAT database.  Additionally, iRODS supports user authentication via PAM (Pluggable Authentication Modules), which can be configured to support many things, including the LDAP or Active Directory (AD) authentication systems.  PAM and TLS have been configured 'available' out of the box with iRODS, but there is still some setup required to configure an installation to communicate with your external authentication server of choice.
 
 The iRODS administrator can 'force' a particular authentication scheme for a rodsuser by 'blanking' the native password for the rodsuser.  There is currently no way to signal to a particular login attempt that it is using an incorrect scheme ([GitHub Issue #2005](https://github.com/irods/irods/issues/2005)).
 
+As of iRODS 5.1.0, this authentication scheme is considered "legacy". Administrators should plan to transition to "irods" authentication.
+
+## `irods` Authentication
+
+The `irods` authentication scheme is a built-in, password-based authentication mechanism which uses time-limited session tokens to authenticate user connections. It is intended to replace the legacy `native` authentication scheme in a future iRODS release.
+
+This scheme strengthens security, introduces configurable password hashing, and uses OAuth2-style session tokens for subsequent authentications after the initial password-based login.
+
+The `irods` authentication scheme is available in iRODS 5.1.0 and later.
+
+This is the basic flow of `irods` authentication:
+
+1. The client sends a password to the server along with information about the user being authenticated.
+1. The server hashes the password and checks it against the one stored in the catalog.
+1. If successful, a session token is generated and returned to the client.
+1. The client uses the session token to authenticate this connection and any subsequent connections.
+1. Once the session token expires, the client must authenticate using a password to get a new session token.
+
+### How to use
+
+In order to use `irods` authentication, iCommands users need to set their authentication scheme to "irods" by including this line in the client environment file:
+```json
+"irods_authentication_scheme": "irods"
+```
+One can also use the `--prompt-auth-scheme` option of `iinit` to provide the authentication scheme name.
+
+TLS is required in order to use `irods` authentication. See [Client TLS Setup](../../system_overview/tls/#client-tls-setup) for more information on how to set up TLS in a client environment.
```

Comment:
This feels it could be decorated with `!!! Note`, but I get that using the styling can get out of hand lol.

Reply:
Done